lib/queues.js

```diff
@@ -163,15 +164,24 @@ class Queue {
   // Push job to queue with the given delay.  Delay can be millisecond,
   // or a string of the form '5s', '2m', etc.
   delayJob(job, duration) {
+    const self = this;
+
     return this.delayJobs([ job ], duration)
-      .then(jobIDs => jobIDs[0]);
+      .then(function(jobIDs) {
+
+        if (process.env.NODE_ENV === 'test')
+          self._delayedJobs.set(jobIDs[0], Date.now() + ms(duration));
```

Comment:
I thought about moving this to delayJob, but `delayJobs` isn't an exposed method, and a value for the duration is only passed to `delayJob`. so why go through the trouble of setting a job id/duration in the map for jobs with no duration / no delay. 🤔